app/src/main/java/com/github/sdpteam15/polyevents/fragments/ProfileFragment.kt

```diff
@@ -88,6 +101,102 @@ class ProfileFragment : Fragment() {
                     }
                 }
         }
+
+        initProfileList(viewRoot)
+
         return viewRoot
     }
+
+    fun initProfileList(viewRoot: View) {
+        profiles.observeRemove {
+            if (it.sender != currentDatabase)
+                currentDatabase.removeProfile(it.value)
+        }
+        profiles.observeAdd {
+            if (it.sender != currentDatabase)
+                currentDatabase.addUserProfileAndAddToUser(it.value, currentUser!!)
+        }
+
+        recyclerView = viewRoot.findViewById(R.id.id_recycler_profile_list)
+        recyclerView.adapter = ProfileAdapter(this, profiles)
+
+        if(currentUser!!.profiles.size > 0)
+            currentDatabase.getUserProfilesList(profiles, currentUser!!).observe(this) {
+                if (!it.value)
+                    HelperFunctions.showToast(getString(R.string.fail_to_update), activity)
+            }
+
+        viewRoot.findViewById<ImageButton>(R.id.id_add_profile_button)
+            .setOnClickListener { createProfilePopup() }
+    }
+
+    private fun createProfilePopup() {
+        // Initialize a new layout inflater instance
+        val inflater: LayoutInflater =
+            (activity)!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+
+        // Inflate a custom view using layout inflater
+        val view = inflater.inflate(R.layout.popup_profile, null)
+
+        // Initialize a new instance of popup window
+        val popupWindow = PopupWindow(
+            view, // Custom view to show in popup window
+            LinearLayout.LayoutParams.MATCH_PARENT, // Width of popup window
+            LinearLayout.LayoutParams.WRAP_CONTENT // Window height
+        )
+
+        val slideIn = Slide()
+        slideIn.slideEdge = Gravity.TOP
+        popupWindow.enterTransition = slideIn
+
+        // Slide animation for popup window exit transition
+        val slideOut = Slide()
+        slideOut.slideEdge = Gravity.END
+        popupWindow.exitTransition = slideOut
+
+
+        // Get the widgets reference from custom view
+        val profileName = view.findViewById<EditText>(R.id.id_edittext_profile_name)
+        val confirmButton = view.findViewById<ImageButton>(R.id.id_confirm_add_item_button)
+
+        //set focus on the popup
+        popupWindow.isFocusable = true
+
+        // Set a click listener for popup's button widget
+        confirmButton.setOnClickListener {
+            profiles.add(
+                UserProfile(
+                    profileName = profileName.text.toString()
+                ), this
+            )
+            // Dismiss the popup window
+            popupWindow.dismiss()
+        }
+
+
+        // Finally, show the popup window on app
+        TransitionManager.beginDelayedTransition(this.recyclerView)
+        popupWindow.showAtLocation(this.recyclerView, Gravity.CENTER, 0, 0)
+    }
```

Comment:
You can keep it like this for now because I'm not sure how to do it in a clean way exactely, but as Cyrille already mentionned on my previous pull request, we should do something for theses long popup methods, maybe make a generic createPopup in HelpersFunction or try to split it in two parts.